lib/surro-gate/proxy.rb

```diff
@@ -111,16 +111,23 @@ def thread_stop
     end
 
     def reactor
-      # Atomically get an array of readable/writable monitors
-      monitors = @mutex.synchronize { @selector.select(0.1) || [] }
+      # Atomically get an array of readable monitors while also polling for writables
+      monitors = @mutex.synchronize do
+        @writer.select(0.1)
+        @reader.select(0.1) || []
+      end
       # Call each transmission proc and collect the results
-      callers = monitors.map { |m| m.value.call }
-      # Sleep for a short time if there was no transmission
-      sleep(0.1) if callers.none? && monitors.any?
+      monitors.map { |m| m.value.call }
+      # Sleep for a short time if there are no sockets
+      sleep(0.1) if @reader.empty? && @writer.empty?
     end
 
     def includes?(*sockets)
-      sockets.map { |socket| @selector.registered?(socket) }.any?
+      sockets.each do |socket|
+        @selectors.each { |selector| return true if selector.registered?(socket) }
+      end
+      # Implicitly return with false if the loop finished
+      false
```

Comment:
Can't this just be a doubly nested `.any?`:

```ruby
sockets.any? do |socket|
  @selectors.any? { |selector| selector.registered?(socket) }
end
```

Avoids the "jump" of `return true`, and I believe that `.any?` will short circuit if `true` is returned anyway.